src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,65 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Status } from '../../Enum/Status';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
+type Props = {
+  filter: Status,
+  setFilter: (filter: Status) => void,
+  query: string,
+  setQuery: (query: string) => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filter, setFilter, query, setQuery,
+}) => {
+  const renderSearchIcon = () => (
+    <span className="icon is-left">
+      <i className="fas fa-magnifying-glass" />
+    </span>
+  );
+
+  const renderClearButton = () => (
+    <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+      <button
```

Comment:
Eslint disabled for jsx-a11y/control-has-associated-label rule. Even though sometimes it is necessary to disable eslint rules, in this case it is better to fix the accessibility issue by adding a meaningful aria-label to the button.